jp/_posts/2009-04-11-gae-hackathon-disc-3-jp.md

```diff
@@ -9,36 +9,48 @@ render_with_liquid: false
+- [GAEのAPI1.2で追加になったCronを使って転置インデックスの更新をしてみる](http://d.hatena.ne.jp/a2c/20090409/1239209449)
+- [GAE ハカソン事前MTG](http://d.hatena.ne.jp/a2c/20090407/1239086203)
+- [GoogleAppEngineのReq/Sec計って見た](http://d.hatena.ne.jp/a2c/20090402/1238608082)
+- [gae上でDataStore使わずにmemcacheで転置インデックス作ってみた。](http://d.hatena.ne.jp/a2c/20090401/1238579242)
+- [GoogleAppEngineのサーバサイドの処理時間をProfileで表示させる為にcProfile使う](http://d.hatena.ne.jp/a2c/20090331/1238464001)
+- [GAEのDatastoreが順調に肥大中](http://d.hatena.ne.jp/a2c/20090326/1238000082)
 
-<div class="codeblock amc_python amc_short"><table><tr class="amc_code_odd"><td class="amc_line"><div class="amc1"></div></td><td>...<br /></td></tr><tr class="amc_code_even"><td class="amc_line"><div class="amc2"></div></td><td><span style="color: black;">Document</span>.<span style="color: black;">all</span><span style="color: black;">&#40;</span><span style="color: black;">&#41;</span>.<span style="color: black;">search</span><span style="color: black;">&#40;</span><span style="color: #dc143c;">keyword</span><span style="color: black;">&#41;</span><br /></td></tr><tr class="amc_code_odd"><td class="amc_line"><div class="amc3"></div></td><td>...</td></tr></table></div>
+<!-- textlint-enable spelling -->
 
-<p>まず、SearchableModelがいろいろ、自分を参照したので、継承するのが難しかったから、別のモジュールにコピーして、forkすることにした。それで、<a href="http://bitbucket.org/a2c/gaehackathon_misopotato/src/e26dda1c611c/saichugen/ian/search/__init__.py#cl-195">この辺</a>に単語の分け方を a2cが作ってくれた <a href="http://bitbucket.org/a2c/gaehackathon_misopotato/src/e26dda1c611c/saichugen/a2c/n_gram.py#cl-1">ngram実装</a>に切り替えた。(<a href="http://d.hatena.ne.jp/keyword/N-gram">ngramとは？</a>)　それで、SearchableModelのモジュールを変えるだけで、googleの実装と同じように使える。</p>
+`SearchableModel`のAPIは基本的にいいと思ったので、`SearchableModel`と同じように、日本語対応できる`SearchableModel`を使いたいなと思いました。こう書けば、わりと簡単に検索できる。
 
-<p>検索キーワードをngramで分けて、インデクスを検索すると、ちゃんと部分的にデータが引っ張ってくる。 <a href="http://saichugen.appspot.com/">http://saichugen.appspot.com/</a> でテストアプリを見れる。コードは<a href="http://bitbucket.org/a2c/gaehackathon_misopotato/">bitbucket</a>で公開されてる。</p>
+```python
+from google.appengine.ext.search import SearchableModel
+from google.appengine.ext.db import *
 
-<p>これから、a2cさんともっと検索結果を取れるようにするのと、インデクスの容量をへらしたりするのと、分け方を自分で実装できるapiを導入したいと思うので、ぜひ期待してください
+class Document(SearchableModel):
+  title = StringProperty(u"Title")
+  text = TextProperty(u"Body Text")
+```
+
+```python
+...
+Document.all().search(keyword)
+...
+```
```

Comment:
These two code blocks could just be combined.

```suggestion
```python
from google.appengine.ext.search import SearchableModel
from google.appengine.ext.db import *

class Document(SearchableModel):
  title = StringProperty(u"Title")
  text = TextProperty(u"Body Text")

...
Document.all().search(keyword)
...
```
```